integrated_channels/moodle/client.py

```diff
@@ -36,10 +51,13 @@ def __init__(self, enterprise_configuration):
         """
         super(MoodleAPIClient, self).__init__(enterprise_configuration)
         self.config = apps.get_app_config('moodle')
+        self.token = self._get_access_token()
```

Comment:
Hmmm. Given the tokens seem to persist on the moodle side, _should_ we be getting a new token every time instead of persisting it? Would we not end up possibly hitting some data limit or just otherwise polluting the web service token screen with 8000 tokens? lol

Reply:
secondly: mostly a note for myself, but if we go this route, I need to update my parameters as it's using `self.enterprise_configuration.api_token` in several spots.

Reply:
So the way I've seen it is it doesn't actually create a new token every time, if you spam this auth endpoint as the same user you'll get the same token returned.

Let me hold off on merging this, I'll see what it's like with a TTL set on token expiry just to verify that's true in that scenario as well, good catch here.

Reply:
oh yeah I forgot it returns the same token. I had that happen too when we tried testing this a while back.

Reply:
I also wonder if this wrapper could just pass token or username/password and the base url in the format of `base_params = {'moodlewsrestformat': 'json', 'wstoken': self.token, 'wsusername', [etc]}` since you might see I have to use those same values in every single request, but that's something I can also play around with outside of this particular pr.